From 26d39efbb931e04a5e95d504c27ede12d0a81c43 Mon Sep 17 00:00:00 2001
From: Masahito Muroi <muroi.masahito@lab.ntt.co.jp>
Date: Fri, 25 Mar 2016 14:06:00 +0900
Subject: [PATCH] Allows DataSource's config field to have not dict type obj

CongressClient expects all datasource driver has dict object in
config field. It raises an error when a datasource doesn't have
any config.

This patch allows config fields to be None object.

Change-Id: I73354f1073f3f814854652eaeaa4b3bbe4bfcf7d
---

diff --git a/congressclient/common/utils.py b/congressclient/common/utils.py
index 9a381e8..b5cedd4 100644
--- a/congressclient/common/utils.py
+++ b/congressclient/common/utils.py
@@ -77,6 +77,8 @@
     :param data: a dict
     :rtype: a string formatted to {a:b, c:d}
     """
+    if not isinstance(data, dict):
+        return str(data)
     return str({str(key): str(value) for key, value in data.items()})
 
 
diff --git a/congressclient/tests/v1/test_datasource.py b/congressclient/tests/v1/test_datasource.py
index a4b0906..abdf14d 100644
--- a/congressclient/tests/v1/test_datasource.py
+++ b/congressclient/tests/v1/test_datasource.py
@@ -30,7 +30,7 @@
                          "name": "my_name",
                          "enabled": "True",
                          "driver": "driver1",
-                         "config": "None"}]
+                         "config": None}]
         }
         lister = mock.Mock(return_value=response)
         self.app.client_manager.congressclient.list_datasources = lister
@@ -42,6 +42,12 @@
         lister.assert_called_with()
         self.assertEqual(['id', 'name', 'enabled', 'driver', 'config'],
                          result[0])
+        for row in result[1]:
+            self.assertEqual(datasource_name, row[0])
+            self.assertEqual("my_name", row[1])
+            self.assertEqual("True", row[2])
+            self.assertEqual("driver1", row[3])
+            self.assertEqual("None", row[4])
 
     def test_list_datasource_output_not_unicode(self):
         # response json string is converted to dict by oslo jsonutils.loads(),
